go/obsclient/authclient.go

```diff
@@ -0,0 +1,78 @@
+package obsclient
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/obscuronet/go-obscuro/go/rpcclientlib"
+)
+
+// AuthObsClient extends the functionality of the ObsClient for all methods that require encryption when communicating with the enclave
+type AuthObsClient struct {
+	ObsClient
+	c rpcclientlib.Client
```

Comment:
Rather than having the client in there twice, would it make more sense to have the client be a public field on `ObsClient`?